### PR TITLE
Fix capacity response in mm-less ingestion

### PR DESCRIPTION
**Pull request by georgew5656**
Fixing bug where capacity for mm-less ingestion was not being properly reported
### Description
This change: https://github.com/georgew5656/druid/commit/3954685aae1f9a78c81bf788e8f69d10324c28d1 to report additional metrics for mm-less ingestion changed the key of the capacity hint returned in getTotalTaskSlotCount and didn't change it in OverlordResource, causing capacity to be returned as 0.

**Initial plan was:**
I decided to just update the hardcoded value in OverlordResource because we can't ref classes from the optional extension in core and I couldn't really find a good constants file to put the constant in for OverlordResource. Open to changing this if someone has a strong opinion here.

After some discussion I decided to move the logic for calculating capacity to the existing totalWorkerCapacity API instead of the /workers api. This cleans up the logic in the workers api (no need to pretend the k8s task runner has a worker), and lets us delete the isK8sTaskRunner method on the taskRunner.

#### Release note
* Fix capacity response in mm-less ingestion.
* Adds a new usedClusterCapacity to the GET /totalWorkerCapacity response. This API should be used to get the total ingestion capacity on the overlord.

##### Key changed/added classes in this PR
 * `OverlordResource`
 * `TaskRunner,RemoteTaskRunner,HttpRemoteTaskRunner,KubernetesTaskRunner`

K8s Task Runner with capacity of 100
Screenshot 2023-08-24 at 9 52 15 AM

Http Remote Task runner with a worker with capacity of 5
Screenshot 2023-08-24 at 9 53 29 AM

This PR has:

- [X] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [] added integration tests.
- [X] been tested in a test Druid cluster.

### CODE REVIEW

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java [hunk 1]
```diff
@@ -947,7 +947,7 @@ public Response apply(TaskRunner taskRunner)
                           "http",
                           "host",
                           "8100",
-                          taskRunner.getTotalTaskSlotCount().getOrDefault("taskQueue", 0L).intValue(),
+                          taskRunner.getTotalTaskSlotCount().getOrDefault("_k8s_worker_category", 0L).intValue(),
```
Comment (kfaraz) [inline indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java:950]: Could you please update the comment above to indicate why this specific value is required? Also, please check if the value can be brought out into a constant somewhere.

**Comment by kfaraz** [comment]
@georgew5656 , @YongGang , I am a little confused on why we should be using this API rather than the getTotalWorkerCapacity API for this purpose.

**Comment by YongGang** [comment]
@kfaraz I believe this is only used by Druid console. This bug caused the following issue when run MSQ.
In the SaaS implementation, this API is not used.
Screenshot 2023-08-18 at 1 54 28 PM

**Review by kfaraz — APPROVED** [review]
Approving for now. The correct fix would be to update the API used by web-console so that we don't need special handling for `K8STaskRunner`.

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java [hunk 2]
```diff
@@ -469,27 +467,17 @@ public Response getWorkerConfig()
...
-    } else {
-      log.debug(
-          "Cannot calculate capacity as task runner [%s] of type [%s] does not support listing workers",
-          taskRunner,
-          taskRunner.getClass().getName()
-      );
-      workers = ImmutableList.of();
-      currentCapacity = -1;
-    }
+    Collection<ImmutableWorkerInfo> workers = taskRunner instanceof WorkerTaskRunner ?
```
Comment (kfaraz) [inline indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java:476]: Do we still need the list of workers?

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskRunner.java [hunk 3]
```diff
@@ -154,5 +146,14 @@ default void updateLocation(Task task, TaskLocation location)
     // do nothing
   }
 
+  default int getTotalCapacity()
```
Comment (kfaraz) [inline indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskRunner.java:149]: Best add javadocs to these methods esp to point out that they can return -1 when not implemented or can't be determined.

**Review by kfaraz — APPROVED** [review]
Thanks for the fix, @georgew5656 , the `TaskRunner` interface makes much more sense now. I have left some minor suggestions.

Could you please share a web-console screenshot after your changes to demonstrate the capacity is being populated correctly?

**Comment by kfaraz** [comment]
@georgew5656 , also, please update the release note to reflect that a new field has been added in the API response and that this API should now be used to get total and used cluster capacity instead of the getWorkers API.

**Pull request merged**